src/main/java/ch/fmi/ContourPolygonTextToBitmask.java

```diff
@@ -0,0 +1,107 @@
+import org.locationtech.jts.geom.GeometryFactory;
+import org.locationtech.jts.geom.Polygon;
+import org.locationtech.jts.io.ParseException;
+import org.locationtech.jts.io.WKTReader;
+import org.scijava.ItemIO;
+import org.scijava.command.Command;
+import org.scijava.command.ContextCommand;
+import org.scijava.log.LogService;
+import org.scijava.plugin.Parameter;
+import org.scijava.plugin.Plugin;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Plugin(type = Command.class, headless = true, menuPath = "FMI>Text To 2D Binary Mask")
+public class ContourPolygonTextToBitmask extends ContextCommand {
+
+    @Parameter
+    private LogService logService;
+
+    @Parameter(label = "Well-Known Text Polygons")
+    private String text;
+
+    @Parameter(type = ItemIO.OUTPUT)
+    private Img<BitType> output;
+
+    private FinalInterval envelopeToDims(Geometry envelope) throws Exception {
+        Coordinate boxMin, boxMax;
+        boxMin = envelope.getCoordinates()[0];
+        if(envelope.getDimension() == 2) boxMax = envelope.getCoordinates()[2];
+        else if(envelope.getDimension() == 1) boxMax = envelope.getCoordinates()[1];
+        else throw new Exception(
+                String.format("Your envelop <%s> stems from another world!", envelope.toText()));
+        long[] min = new long[] { (long) boxMin.getX(), (long) boxMin.getY() };
+        long[] max = new long[] { (long) boxMax.getX(), (long) boxMax.getY() };
+        return new FinalInterval(min, max);
+    }
+
+    @Override
+    public void run() {
+        GeometryFactory factory = new GeometryFactory();
+        WKTReader reader = new WKTReader(factory);
+        try {
+            // get GIS polygon from input
+            Polygon polygon = (Polygon) reader.read(text);
+            // construct bitmask
+            List<RealLocalizable> points = Arrays.stream(polygon.getCoordinates())
+                    .map(coordinate -> new RealPoint(coordinate.getX(), coordinate.getY()))
+                    .collect(Collectors.toList());
+            Polygon2D polygon2d = GeomMasks.polygon2D(points);  // imglib2-roi polygon
+            RealRandomAccessibleRealInterval<BoolType> bitmask = Masks.toRealRandomAccessibleRealInterval(polygon2d);
+            // create view
+            FinalInterval dims = envelopeToDims(polygon.getEnvelope());
+            IntervalView<BoolType> view = Views.interval(Views.raster(bitmask), dims);
+            // create image
+            RandomAccessibleInterval<BitType> interval = Converters.convert(
+                    (RandomAccessibleInterval<BoolType>) view, (in, out) -> out.set(in.get()), new BitType());
+            Img<BitType> image = new ImgView<>(interval, new ArrayImgFactory<>());
+            // return result
+            output =  image;
+        } catch (ParseException e) {
+            logService.warn(String.format("Could not create a polygon from input <{}>", text));
+        } catch(Exception e) {
```

Comment:
Can we catch a more specific `Exception` here?

Reply:
Yeah sure, I added an `IllegalArgumentException`.